06.wc/my_wc.rb

```diff
@@ -29,16 +28,28 @@ def display_lines_words_bytes(options, contents)
 end
 
 def adjust_lines_words_bytes(lines, words, bytes, file_or_total, options)
-  options['l'] ? lines.to_s.rjust(8) + " #{file_or_total}" : "#{lines.to_s.rjust(8)}#{words.to_s.rjust(8)}#{bytes.to_s.rjust(8)}" + " #{file_or_total}"
+  if options['l']
+    "#{apply_to_s_rjust(lines)} #{file_or_total}"
+  else
+    "#{apply_to_s_rjust(lines)}#{apply_to_s_rjust(words)}#{apply_to_s_rjust(bytes)} #{file_or_total}"
+  end
+end
+
+def apply_to_s_rjust(integer)
+  integer.to_s.rjust(8)
 end
 
 def read_contents
   if ARGV.empty?
-    [{ '' => readlines }]
+    { '' => readlines }
   else
-    ARGV.map do |file|
-      { file => File.open(file, &:readlines) }
+    hash = {}
+    ARGV.each do |file|
+      File.open(file) do |f|
+        hash[f.path] = f.readlines
+      end
     end
+    hash
```

Comment:
高等テクニックですが、こんな感じでも動くかも？

```suggestion
    ARGV.to_h do |file|
      [file, File.readlines(file)]
    end
```